### PR TITLE
feat: Datepicker Calendar

### PR DESCRIPTION
**Pull request by bgutsol**
# Purpose of PR

- Calendar component for Datepicker

### CODE REVIEW

File: scripts/.storybook/main.js [hunk 1]
```diff
@@ -18,9 +18,9 @@ module.exports = {
 
   webpackFinal: async (config, { configType }) => {
     // Need to remove Storybook's CSS loader before adding our own
-    config.module.rules = config.module.rules.filter(
-      (rule) => rule.test.toString() !== '/\\.css$/',
-    );
+    // config.module.rules = config.module.rules.filter(
+    //   (rule) => rule.test.toString() !== '/\\.css$/',
+    // );
```
Comment (denkristoffer) [inline scripts/.storybook/main.js:24]: We should just delete this.

File: packages/components/datepicker/stories/Calendar.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  );
+};
+
+export const WithFromToDates: Story<CalendarProps> = (args) => {
+  const [selectedDay, setSelectedDay] = useState<Date>(new Date());
+
+  return (
+    <Calendar
+      {...args}
+      mode="single"
```
Comment (denkristoffer) [inline packages/components/datepicker/stories/Calendar.stories.tsx:32]: Do we want to support other modes, or should we omit this prop from `CalendarProps`?

**Review by denkristoffer — CHANGES_REQUESTED** [review]
The styling looks off to me, I made a comment on Chromatic: https://www.chromatic.com/component?appId=5fd1dda724cc620021ace8c5&csfId=components-datepicker-calendar&specName=With+From+To+Dates&buildNumber=3166&k=62964ca2bec7e5004a48eb4c-1200-interactive-true&h=3&b=-1

File: packages/components/datepicker/stories/Calendar.stories.tsx (hunk 2 above)
Comment (bgutsol) [reply packages/components/datepicker/stories/Calendar.stories.tsx:32]: I would support it, don't see any reason to restrict it.

Comment (denkristoffer) [reply packages/components/datepicker/stories/Calendar.stories.tsx:32]: Nevermind, I see you use both.

**Comment by bgutsol** [comment]
> The styling looks off to me, I made a comment on Chromatic: https://www.chromatic.com/component?appId=5fd1dda724cc620021ace8c5&csfId=components-datepicker-calendar&specName=With+From+To+Dates&buildNumber=3166&k=62964ca2bec7e5004a48eb4c-1200-interactive-true&h=3&b=-1

Pls, check this one more time when the build will be updated with the recent styles changes

**Review by denkristoffer — CHANGES_REQUESTED** [review]
Added a few notes.

Also, I've asked writing team if we want Sunday or Monday as the first day of the week.

**Comment by denkristoffer** [comment]
Feedback from technical writing is that we should go with Monday as the first day of the week 🎉

**Comment by bgutsol** [comment]
> Feedback from technical writing is that we should go with Monday as the first day of the week 🎉

Got it, thanks Kristoffer! I’m gonna make Monday as a first day of the week by default but still leave an option to configure it

File: packages/components/datepicker/src/Calendar/CalendarCaptionLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  return (
+    <Heading
+      as="h2"
+      aria-live="polite"
+      aria-atomic="true"
+      id={props.id}
+      className={cx(styles.caption_label, isVisuallyHidden && styles.vhidden)}
+    >
+      {formatCaption(props.displayMonth, { locale })}
```
Comment (mshaaban0) [inline packages/components/datepicker/src/Calendar/CalendarCaptionLabel.tsx:28]: Should we include the year there as well?

Screenshot 2022-06-02 at 17 14 02

File: packages/components/datepicker/stories/Calendar.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import { Calendar } from '../src';
+import type { CalendarProps } from '../src';
+import { DateRange } from 'react-day-picker';
+
+export default {
+  component: Calendar,
+  title: 'Components/Datepicker/Calendar',
+} as Meta;
+
+export const Default: Story<CalendarProps> = (args) => {
```
Comment (mshaaban0) [inline packages/components/datepicker/stories/Calendar.stories.tsx:13]: I think the website page will also need some love and good examples

File: packages/components/datepicker/src/Calendar/Calendar.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+// Props comparison to our current datepickers:
+// Current      DayPicker
+// value        selected
+// onChange     onSelect
+// min          fromDate/fromMonth/fromYear
+// max          toDate/toMonth/toYear
+
+export type CalendarProps = DayPickerProps;
+
+export function Calendar(props: CalendarProps) {
```
Comment (mshaaban0) [inline packages/components/datepicker/src/Calendar/Calendar.tsx:18]: Can we use jsdoc here? some info with a link to DayPickerProps docs will be cool

File: packages/components/datepicker/stories/Calendar.stories.tsx (hunk 4 above)
Comment (mshaaban0) [reply packages/components/datepicker/stories/Calendar.stories.tsx:13]: Screenshot 2022-06-02 at 16 58 14
Menu structure should be under date components section

File: packages/components/datepicker/src/index.ts [hunk 6]
```diff
@@ -1,2 +1,4 @@
 export { Datepicker } from './Datepicker';
 export type { DatepickerProps } from './Datepicker';
+
+export * from './Calendar';
```
Comment (mshaaban0) [inline packages/components/datepicker/src/index.ts:4]: I'm confused a bit about `Datepicker` and `Calendar` What's the difference?
Would you be able to provide an example of user's import use cases?

**Review by bgutsol — COMMENTED** [review]
Feedback from @domarku 
- Omit Range variant and delete it completely;
- Rename "With From To Dates" to "With date boundaries" or "min and max dates";
- Make Date buttons fully rounded;
- Maybe provide `minDate=today` as a default prop?? **(Discuss with the team)**

File: packages/components/datepicker/src/index.ts (hunk 6 above)
Comment (bgutsol) [reply packages/components/datepicker/src/index.ts:4]: Datepicker is a combobox, will be done in the follow up stories.

File: packages/components/datepicker/stories/Calendar.stories.tsx (hunk 4 above)
Comment (bgutsol) [reply packages/components/datepicker/stories/Calendar.stories.tsx:13]: will be done in the separate documentation story

File: packages/components/datepicker/src/Calendar/CalendarCaptionLabel.tsx (hunk 3 above)
Comment (bgutsol) [reply packages/components/datepicker/src/Calendar/CalendarCaptionLabel.tsx:28]: I don't think it's necessary. When you focus calendar with VoiceOver it tells you what month and year it is, then it tells you the chosen date. The [W3C](https://w3c.github.io/aria-practices/examples/dialog-modal/datepicker-dialog.html) example just reads numbers when you navigate between dates.

TLDR: if it was easy to change I would do it, but we will have to customize the whole Day component. So it doesn't really const the efforts, IMO

File: packages/components/datepicker/src/Calendar/Calendar.styles.ts [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+      '.rdp-multiple_months .rdp-caption_end &': {
+        position: 'absolute',
+        top: '50%',
+        right: '0',
+        transform: 'translateY(-50%)',
+      },
+    }),
+
+    nav_button: css({
+      width: '32px',
```
[the remark this answers is not in the document]
Comment (bgutsol) [reply packages/components/datepicker/src/Calendar/Calendar.styles.ts:124]: I don't think we have size tokens, but I've changed it to `rem`

[the remark this answers is not in the document]
Comment (denkristoffer) [reply packages/components/datepicker/src/Calendar/Calendar.styles.ts]: I see what you mean, but we've been using the spacing tokens for that as well 🤔

**Review by denkristoffer — APPROVED** [review]
Great work! 😍

**Comment by domarku** [comment]
> Feedback from @domarku
> 
> * Remove range variant from storybook;
> * Rename "With From To Dates" to "With date boundaries" or "min and max dates";
> * Make Date buttons fully rounded;
> * Maybe provide `minDate=today` as a default prop?? **(Discuss with the team)**

Hey @bgutsol I would add one more thing:
- Mark today with `--blue-600` and `--font-weight-demi-bold`

File: packages/components/datepicker/src/Calendar/Calendar.styles.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (bgutsol) [reply packages/components/datepicker/src/Calendar/Calendar.styles.ts]: ah, I see. I don't have any strong opinion on it, if you have - I can do it in another PR
